github/data_source_github_organization_teams.go

```diff
@@ -0,0 +1,136 @@
+package github
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/shurcooL/githubv4"
+	"log"
+	"strconv"
+	"time"
+)
+
+func dataSourceGithubOrganizationTeams() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGithubOrganizationTeamsRead,
+
+		Schema: map[string]*schema.Schema{
+			"root_teams_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"teams": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"node_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"slug": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"privacy": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"members": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGithubOrganizationTeamsRead(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	client := meta.(*Owner).v4client
+	orgName := meta.(*Owner).name
+	rootTeamsOnly := d.Get("root_teams_only").(bool)
+
+	log.Print("[INFO] Refreshing GitHub Teams for Organization: ", orgName)
+
+	var query TeamsQuery
+	variables := map[string]interface{}{
+		"first":         githubv4.Int(100),
+		"login":         githubv4.String(orgName),
+		"cursor":        (*githubv4.String)(nil),
+		"rootTeamsOnly": githubv4.Boolean(rootTeamsOnly),
+	}
+
+	var teams []interface{}
+	for {
+		err = client.Query(meta.(*Owner).StopContext, &query, variables)
+		if err != nil {
+			return err
+		}
+
+		additionalTeams := flattenGitHubTeams(query)
+		teams = append(teams, additionalTeams...)
+
+		if !query.Organization.Teams.PageInfo.HasNextPage {
+			break
+		}
+		variables["cursor"] = githubv4.NewString(query.Organization.Teams.PageInfo.EndCursor)
+	}
+
+	d.SetId(strconv.FormatInt(time.Now().Unix(), 10))
```

Comment:
Interesting approach here.  🤔  I'm unsure if this will cause unwanted behaviour and have not seen this pattern used before.  👍🏾  on trying it but we'll need to keep an eye out for fixes that may need to occur due to choosing what seems to be a non-deterministic ID.  

I see a similar data source pins to the overall org ID:

```go
// github/data_source_github_organization.go
d.SetId(strconv.FormatInt(organization.GetID(), 10))
```

Reply:
Sorry, added another change. In the previous commit I added the value of `root_teams_only` to the ID so it would be unique between multiple instances of the data source (edge case). But that is of course not necessary, so I simplified that again.